test/e2e/resource/create_test.go

```diff
@@ -0,0 +1,704 @@
+					return err
+				}
+
+				if _, err := c.Send(string(terminal.KeyArrowDown)); err != nil {
+					return err
+				}
+
+				if _, err := c.ExpectString("image"); err != nil {
+					return err
+				}
+
+				if _, err := c.Send(string(terminal.KeyEnter)); err != nil {
+					return err
+				}
+
+				if _, err := c.ExpectString("Enter a value for url :"); err != nil {
+					return err
+				}
+
+				if _, err := c.Send("quay.io/tekton/controller"); err != nil {
+					return err
+				}
+
+				if _, err := c.Send(string(terminal.KeyEnter)); err != nil {
+					return err
+				}
+				if _, err := c.ExpectString("Enter a value for digest :"); err != nil {
+					return err
+				}
+
+				if _, err := c.Send(string(terminal.KeyEnter)); err != nil {
+					return err
+				}
+
+				if _, err := c.ExpectString("New image resource \"my-image\" has been created"); err != nil {
+					return err
+				}
+
+				if _, err := c.ExpectEOF(); err != nil {
+					return err
+				}
+
+				c.Close()
+				return nil
+			}})
+
+	})
+
+	t.Run("list single pipeline resource", func(t *testing.T) {
+		res := tkn.Run("resource", "list")
+		e2e.Assert(t, res, icmd.Success)
+		golden.Assert(t, res.Stdout(), strings.ReplaceAll(fmt.Sprintf("%s.golden", t.Name()), "/", "-"))
+	})
+}
+
+func TestCreateCloudEventResourceInteractively(t *testing.T) {
+	t.Parallel()
+	c, namespace := e2e.Setup(t)
+	knativetest.CleanupOnInterrupt(func() { e2e.TearDown(t, c, namespace) }, t.Logf)
+	defer e2e.TearDown(t, c, namespace)
+
+	tkn, err := e2e.NewTknRunner(namespace)
+	if err != nil {
+		t.Fatalf("Error creating tknRunner %+v", err)
+	}
+
+	t.Run("Create pipeline resource of cloud event type, interactively in namespace "+namespace, func(t *testing.T) {
+		tkn.RunInteractiveTests(t, &e2e.Prompt{
+			CmdArgs: []string{"resource", "create"},
+			Procedure: func(c *expect.Console) error {
+				if _, err := c.ExpectString("Enter a name for a pipeline resource :"); err != nil {
+					return err
+				}
+
+				if _, err := c.Send("my-cloud"); err != nil {
+					return err
+				}
+
+				if _, err := c.Send(string(terminal.KeyEnter)); err != nil {
+					return err
+				}
+
+				if _, err := c.ExpectString("Select a resource type to create :"); err != nil {
+					return err
+				}
+
+				if _, err := c.ExpectString("cloudEvent"); err != nil {
+					return err
+				}
+
+				if _, err := c.Send(string(terminal.KeyEnter)); err != nil {
+					return err
+				}
+
+				if _, err := c.ExpectString("Enter a value for targetURI :"); err != nil {
+					return err
+				}
+
+				if _, err := c.SendLine("http://localhost:8080"); err != nil {
+					return err
+				}
+
+				if _, err := c.ExpectString("New cloudEvent resource \"my-cloud\" has been created"); err != nil {
+					return err
+				}
+
+				if _, err := c.ExpectEOF(); err != nil {
+					return err
+				}
+
+				c.Close()
+				return nil
+			}})
+	})
+
+	t.Run("list single pipeline resource", func(t *testing.T) {
```

Comment:
I would change the name to something more descriptive about which PipelineResource.